### PR TITLE
Validate OBM APIs and Node Patch API 

### PR DESCRIPTION
**Pull request by uppalk1**
@RackHD/corecommitters @jlongever @zyoung51 @keedya @tannoa2

### CODE REVIEW

File: test/modules/obm.py [hunk 1]
```diff
@@ -61,12 +61,15 @@ def setup_nodes(self, service_type='ipmi-obm-service',uuid=None):
...
-                    if self._set_snmp(uid) == False:
-                        err.append('Error setting SNMP OBM settings for node {0}'.format(uid))
+                    if service_type == 'ipmi-obm-service' and node_type == 'compute':
+                        if len(self.check_nodes(uuid=uuid)) > 0:
+                            if self._set_ipmi(uid) == False:
+                                err.append('Error setting IPMI OBM settings for node {0}'.format(uid))
+                            else:
+                                LOG.info('Successful')
+                    if service_type == 'snmp-obm-service' and node_type != 'enclosure':
+                        if self._set_snmp(uid) == False:
```
Comment (jlongever) [inline test/modules/obm.py]: Check for an existing snmp service type as well.

File: test/modules/obm.py [hunk 2]
```diff
@@ -9,6 +9,8 @@
 """
 Class to abstract the RackHD Out-of-Band settings 
 """
+
+
```
Comment (anhou) [inline test/modules/obm.py]: don't need these two blank lines here

**Comment by uppalk1** [comment]
Updating the PR with a new test file config_test.py to validate Config APIs

@RackHD/corecommitters @keedya @tannoa2 @jlongever @zyoung51

**Comment by jlongever** [comment]
UNSTABLE -

**Comment by jlongever** [comment]
+0

**Comment by jlongever** [comment]
please test this

**Comment by jlongever** [comment]
please test this

**Comment by jlongever** [comment]
test this please

**Comment by anhou** [comment]
@jlongever it seems that when Jenkins fail ,there's no links to go to the fail messages at present?

**Comment by jlongever** [comment]
@anhou I wasn't planning on providing a link to the internal Jenkins system,  however Jenkins will post a PR comment on Failure/Unstable status with the failed tests, error details and a stacktrace.   See this PR as an example of what to expect: https://github.com/jlongever/RackHD/pull/6.

**Comment by anhou** [comment]
@jlongever It's great, Thanks!

**Comment by jlongever** [comment]
+1

**Pull request merged**